presto-main/src/main/java/com/facebook/presto/cost/JoinStatsRule.java

```diff
@@ -241,15 +242,10 @@ PlanNodeStatsEstimate calculateJoinComplementStats(
             PlanNodeStatsEstimate leftStats,
             PlanNodeStatsEstimate rightStats)
     {
-        // TODO: add support for non-equality conditions (e.g: <=, !=, >)
-        if (filter.isPresent()) {
-            // non-equi filters are not supported
-            return UNKNOWN_STATS;
-        }
-
         if (criteria.isEmpty()) {
             if (rightStats.getOutputRowCount() > 0) {
                 // all left side rows are matched
+                // TODO: account non-equi conditions here somehow
                 return leftStats.mapOutputRowCount(rowCount -> 0.0);
             }
             if (rightStats.getOutputRowCount() == 0) {
```

Comment:
This (`if right is empty then outer == left side`) is universally true, should be pulled up, before `if (criteria.isEmpty())` condition.